interactive_engine/executor/ir/integrated/tests/pathxd_test.rs

```diff
@@ -625,4 +625,223 @@ mod test {
     fn path_expand_with_filter_query_w2_test() {
         path_expand_with_filter_query(2)
     }
+
+    // both(1..3) with vertices and edges preserved in the path
+    fn path_expand_whole_v_e_query(worker_num: u32) {
+        initialize();
+        let request = init_path_expand_request(pb::Range { lower: 1, upper: 3 }, 2, 0);
+        let mut results = submit_query(request, worker_num);
+        let mut result_collection: Vec<Vec<String>> = vec![];
+        let mut expected_result_paths: Vec<Vec<String>> = vec![
+            // v1, e[1->2], v2
+            vec!["1", "1, 2", "2"],
+            // v1, e[1->4], v4
+            vec!["1", "1, 4", "4"],
+            // v2, e[1->2], v1
+            vec!["2", "1, 2", "1"],
+            // v4, e[1->4], v1
+            vec!["4", "1, 4", "1"],
+            // v1, e[1->2], v2, e[1->2], v1
+            vec!["1", "1, 2", "2", "1, 2", "1"],
+            // v1, e[1->4], v4, e[1->4], v1
+            vec!["1", "1, 4", "4", "1, 4", "1"],
+            // v2, e[1->2], v1, e[1->2], v2
+            vec!["2", "1, 2", "1", "1, 2", "2"],
+            // v2, e[1->2], v1, e[1->4], v4
+            vec!["2", "1, 2", "1", "1, 4", "4"],
```

Comment:
Ha, if I were you. I would make the string v2, instead of 2, and the edge e[1->2] instead of `1, 2`, and all comments can be removed happily.